### ISSUE TITLE
TB 115 Upgrade - QuickFolders toolbar not displayed

### ISSUE THREAD

**Issue by amfonda** [RealRaven2000/QuickFolders#429]
I've recently upgraded from TB 91.13.1 to TB 115.4.1 and I cannot bring up the QuickFolders toolbar.  The TB menu item View > Toolbars does not have an item listed for QuickFolders.  I have installed the QF v6.3 add-on (premium license validated) and have tried removing the add-on and then adding it again to no avail.  Restarting TB or rebooting does not help.  I've searched this forum and all the on-line FAQ but don't see why this is not working.  What am I missing?

**Comment by RealRaven2000** [comment]
Sometimes you have to restart Thunderbird multiple times for the change to take effect. Can you see the QuickFolders settings dialog in Add-ons Manager?

**Comment by amfonda** [comment]
Yes, I've restarted TB more than 20 times since upgrading.  No changes.  Yes, I can see the QF settings dialog and I've searched for something that might be off.  I saved my QF configuration from before the upgrade but the Load Configuration button does not respond.

**Comment by RealRaven2000** [comment]
> Yes, I've restarted TB more than 20 times since upgrading. No changes. Yes, I can see the QF settings dialog and I've searched for something that might be off. I saved my QF configuration from before the upgrade but the Load Configuration button does not respond.

Loading settings from the actual dialog is broken, (that's a separate issue) it's fixed here:

https://github.com/RealRaven2000/QuickFolders/issues/425

**Comment by amfonda** [comment]
OK, I downloaded and installed the test version in #425, but I see no changes to anything on my issue.  I still cannot load my old configuration.

**Comment by amfonda** [comment]
Yesterday I went backwards to TB 115.0 and QF 6.2 and still had no QuickFolders visible.  I've been at this for several days off/on.

**Comment by RealRaven2000** [comment]
> Yes, I've restarted TB more than 20 times since upgrading. No changes. Yes, I can see the QF settings dialog and I've searched for something that might be off. I saved my QF configuration from before the upgrade but the Load Configuration button does not respond.

Loading settings from the actual dialog is broken, (that's a separate issue) it's fixed here:

https://github.com/RealRaven2000/QuickFolders/issues/425

**Comment by RealRaven2000** [comment]
Can you try add a bigger number in QF Settings / Advanced / init delay. Try 10000 for 10 Seconds of waiting... if youy can't fix it a Zoom call to see the behavior might be helpful...

**Comment by RealRaven2000** [comment]
Here is how to create a debug log:

First, open QuickFolders Settings. Then go to the advanced Settings page:
Activate Debug mode (on the black bar on the bottom of advanced settings)
Right-click debug mode, click on the disclaimer (it's safe as it only allows changing relevant debug switches)

Depending on what kind of error you are investigating, find the following additional debug settings and activate them:

| setting | value |
| -- | -- |
| extensions.quickfolders.debug.firstrun | true |
| extensions.quickfolders.debug.interface | true |
| extensions.quickfolders.debug.updateFolders | true |


Open the Errors console in Thunderbird's main menu: Tools / Developer Tools / Error Console.
Clear the output

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/a13eb0b2-5493-4b89-ba84-bb4d15a5c7d5)


IMPORTANT: Configure the console as follows:

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/a6ef66fb-99df-4960-aa79-8581f362f755)

Now you're ready for a test - try to reproduce the error in QuickFolders by carrying out only the last actions that lead to the problem. In your case disable QF then enable it again. Clear the complete output before you enable QF.

Right-click on any message, then select Copy all Messages

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/8fd8a1a3-b618-47e8-9cb4-4c1a555dce89)

**Comment by amfonda** [comment]
OK, here's what I got:
```

1699468914350	addons.webextension.quickfolders@curious.be	WARN	Loading extension 'quickfolders@curious.be': Reading manifest: Warning processing version: version must be a version string consisting of at most 4 integers of at most 9 digits without leading zeros, and separated with dots
1699468914350	addons.webextension.quickfolders@curious.be	WARN	Loading extension 'quickfolders@curious.be': Reading manifest: Warning processing permissions: Error processing permissions.8: Value "sessions" must either: must either [must either [must either [be one of ["idle"], be one of ["menus.overrideContext"], be one of ["activeTab"], be one of ["cookies"], be one of ["scripting"], or be one of ["webRequest", "webRequestBlocking", "webRequestFilterResponse", "webRequestFilterResponse.serviceWorkerScript"]], must either [be one of ["mozillaAddons"], be one of ["activityLog"], be one of ["networkStatus"], or be one of ["telemetry"]], be one of ["alarms", "storage", "unlimitedStorage"], be one of ["geckoProfiler"], be one of ["identity"], be one of ["menus"], be one of ["contextualIdentities"], be one of ["declarativeNetRequestWithHostAccess"], be one of ["dns"], or be one of ["theme"]], must either [must either [be one of ["idle"], be one of ["menus.overrideContext"], be one of ["activeTab"], be one of ["cookies"], be one of ["scripting"], or be one of ["webRequest", "webRequestBlocking", "webRequestFilterResponse", "webRequestFilterResponse.serviceWorkerScript"]], be one of ["clipboardRead", "clipboardWrite", "geolocation", "notifications"], be one of ["accountsRead"], be one of ["addressBooks", "sensitiveDataUpload"], be one of ["browsingData"], be one of ["compose", "compose.save", "compose.send"], be one of ["messagesModify", "sensitiveDataUpload"], be one of ["accountsFolders"], be one of ["accountsIdentities"], be one of ["messagesDelete", "messagesImport", "messagesMove", "messagesRead", "messagesTags", "sensitiveDataUpload"], be one of ["pkcs11"], be one of ["tabs", "tabHide"], be one of ["browserSettings"], be one of ["declarativeNetRequestFeedback"], be one of ["downloads", "downloads.open"], be one of ["management"], be one of ["privacy"], be one of ["proxy"], be one of ["nativeMessaging"], or be one of ["webNavigation"]], be one of ["declarativeNetRequest"], or match the pattern /^experiments(\.\w+)+$/], or must either [be one of ["<all_urls>"], must either [match the pattern /^(https?|wss?|file|ftp|\*):\/\/(\*|\*\.[^*/]+|[^*/]+)\/.*$/, or match the pattern /^file:\/\/\/.*$/], or match the pattern /^resource:\/\/(\*|\*\.[^*/]+|[^*/]+)\/.*$|^about:/]
1699468914394	addons.webextension.quickfolders@curious.be	WARN	Loading extension 'quickfolders@curious.be': Reading manifest: Warning processing version: version must be a version string consisting of at most 4 integers of at most 9 digits without leading zeros, and separated with dots
1699468914395	addons.webextension.quickfolders@curious.be	WARN	Loading extension 'quickfolders@curious.be': Reading manifest: Warning processing permissions: Error processing permissions.8: Value "sessions" must either: must either [must either [must either [be one of ["idle"], be one of ["menus.overrideContext"], be one of ["activeTab"], be one of ["cookies"], be one of ["scripting"], or be one of ["webRequest", "webRequestBlocking", "webRequestFilterResponse", "webRequestFilterResponse.serviceWorkerScript"]], must either [be one of ["mozillaAddons"], be one of ["activityLog"], be one of ["networkStatus"], or be one of ["telemetry"]], be one of ["alarms", "storage", "unlimitedStorage"], be one of ["geckoProfiler"], be one of ["identity"], be one of ["menus"], be one of ["contextualIdentities"], be one of ["declarativeNetRequestWithHostAccess"], be one of ["dns"], or be one of ["theme"]], must either [must either [be one of ["idle"], be one of ["menus.overrideContext"], be one of ["activeTab"], be one of ["cookies"], be one of ["scripting"], or be one of ["webRequest", "webRequestBlocking", "webRequestFilterResponse", "webRequestFilterResponse.serviceWorkerScript"]], be one of ["clipboardRead", "clipboardWrite", "geolocation", "notifications"], be one of ["accountsRead"], be one of ["addressBooks", "sensitiveDataUpload"], be one of ["browsingData"], be one of ["compose", "compose.save", "compose.send"], be one of ["messagesModify", "sensitiveDataUpload"], be one of ["accountsFolders"], be one of ["accountsIdentities"], be one of ["messagesDelete", "messagesImport", "messagesMove", "messagesRead", "messagesTags", "sensitiveDataUpload"], be one of ["pkcs11"], be one of ["tabs", "tabHide"], be one of ["browserSettings"], be one of ["declarativeNetRequestFeedback"], be one of ["downloads", "downloads.open"], be one of ["management"], be one of ["privacy"], be one of ["proxy"], be one of ["nativeMessaging"], or be one of ["webNavigation"]], be one of ["declarativeNetRequest"], or match the pattern /^experiments(\.\w+)+$/], or must either [be one of ["<all_urls>"], must either [match the pattern /^(https?|wss?|file|ftp|\*):\/\/(\*|\*\.[^*/]+|[^*/]+)\/.*$/, or match the pattern /^file:\/\/\/.*$/], or match the pattern /^resource:\/\/(\*|\*\.[^*/]+|[^*/]+)\/.*$|^about:/]
Finished setting up license startup code qf-background.js:243:24
1699468918581	addons.webextension.quickfolders@curious.be	WARN	Loading extension 'quickfolders@curious.be': Reading manifest: Warning processing version: version must be a version string consisting of at most 4 integers of at most 9 digits without leading zeros, and separated with dots
1699468918582	addons.webextension.quickfolders@curious.be	WARN	Loading extension 'quickfolders@curious.be': Reading manifest: Warning processing permissions: Error processing permissions.8: Value "sessions" must either: must either [must either [must either [be one of ["idle"], be one of ["menus.overrideContext"], be one of ["activeTab"], be one of ["cookies"], be one of ["scripting"], or be one of ["webRequest", "webRequestBlocking", "webRequestFilterResponse", "webRequestFilterResponse.serviceWorkerScript"]], must either [be one of ["mozillaAddons"], be one of ["activityLog"], be one of ["networkStatus"], or be one of ["telemetry"]], be one of ["alarms", "storage", "unlimitedStorage"], be one of ["geckoProfiler"], be one of ["identity"], be one of ["menus"], be one of ["contextualIdentities"], be one of ["declarativeNetRequestWithHostAccess"], be one of ["dns"], or be one of ["theme"]], must either [must either [be one of ["idle"], be one of ["menus.overrideContext"], be one of ["activeTab"], be one of ["cookies"], be one of ["scripting"], or be one of ["webRequest", "webRequestBlocking", "webRequestFilterResponse", "webRequestFilterResponse.serviceWorkerScript"]], be one of ["clipboardRead", "clipboardWrite", "geolocation", "notifications"], be one of ["accountsRead"], be one of ["addressBooks", "sensitiveDataUpload"], be one of ["browsingData"], be one of ["compose", "compose.save", "compose.send"], be one of ["messagesModify", "sensitiveDataUpload"], be one of ["accountsFolders"], be one of ["accountsIdentities"], be one of ["messagesDelete", "messagesImport", "messagesMove", "messagesRead", "messagesTags", "sensitiveDataUpload"], be one of ["pkcs11"], be one of ["tabs", "tabHide"], be one of ["browserSettings"], be one of ["declarativeNetRequestFeedback"], be one of ["downloads", "downloads.open"], be one of ["management"], be one of ["privacy"], be one of ["proxy"], be one of ["nativeMessaging"], or be one of ["webNavigation"]], be one of ["declarativeNetRequest"], or match the pattern /^experiments(\.\w+)+$/], or must either [be one of ["<all_urls>"], must either [match the pattern /^(https?|wss?|file|ftp|\*):\/\/(\*|\*\.[^*/]+|[^*/]+)\/.*$/, or match the pattern /^file:\/\/\/.*$/], or match the pattern /^resource:\/\/(\*|\*\.[^*/]+|[^*/]+)\/.*$|^about:/]

```

**Comment by RealRaven2000** [comment]
that's not an awful lot... the prerelease versions should work anyway. At least it seems like the main() function of the background page is executed. But there should be a load more messages afterwards.,..

In main() there should also be a line
QuickFolders: add toggle-foldertree command... 

are there no errors or warnings apart from the manifest warnings that start with "1699..." ?

 (I think these are partly from Firefox and may not even apply to Thunderbird Add-ons, but I need to confirm with @jobisoft  from Thunderbird)

**Comment by amfonda** [comment]
I ran the debug routine again and got the same output.  I'm not sure you're instructing me to do something else?

**Comment by RealRaven2000** [comment]
> I ran the debug routine again and got the same output. I'm not sure you're instructing me to do something else?

Not really. I trust you have enabled errors warnings and logs, then there should be a lot more activity. I will be out of office until Sunday, so i am going to ask John ( i think he created WindowListener) early next week…

**Comment by amfonda** [comment]
Yes, I enabled errors, warnings and logs.

Well I found something that works.  I added a new mail account to Thunderbird and all of a sudden the QuickFolders toolbar appears.  I did nothing else.  I was able to load my old configuration and it looks fine except I want to center my tabs in the toolbar as I had done before.  I don't see any option for that.

I am going to try and restart TB to see it this current state holds...  NO, Quickfolders are now gone again.  Drats!

**Comment by RealRaven2000** [comment]
> Well I found something that works. I added a new mail account to Thunderbird and all of a sudden the QuickFolders toolbar appears. I did nothing else. I was able to load my old configuration and it looks fine except I want to center my tabs in the toolbar as I had done before. I don't see any option for that.
> 
> I am going to try and restart TB to see it this current state holds... NO, Quickfolders are now gone again. Drats!

Hah you jogged my memory. There may be a bug in the accounts API that van lead to crashes when there is an invalid account. My license code iterates all accounts in order to validate the license, maybe there is an old broken mail account that triggers the problem. 

Most radical solution would be to create a new profile and then import the accounts one by one… but it would also guarantee much smoother operation for the future.

**Comment by amfonda** [comment]
Yeah, I can try that... first I'll try to remove/restore individual mail accounts in the same profile.

**Comment by amfonda** [comment]
How do I center my QF tabs within the toolbar?

**Comment by RealRaven2000** [comment]
Peobably

> How do I center my QF tabs within the toolbar?

Please open a new issue you can tag it as question if you like.

**Comment by amfonda** [comment]
I created a new profile in Thunderbird... I added all my mail accounts... I closed TB in order to manually copy over my mail and calendar data... I restarted TB, then I added QuickFolders 6.3 add-on... and NO Quickfolders Toolbar is visible or shown in the Menu > View Toolbars list.

I tried the same trick and added another mail account... and the QuickFolder toolbar magically appears.  Then I restarted TB and its gone... the exact same behavior as my original profile... which I will now switch back to.

**Comment by amfonda** [comment]
Well I found another temporary fix... if I go to Thunderbird Account Settings > Account Actions > Add Mail Account... then cancel... the QuickFolders Toolbar miraculously appears... temporarily, until I close/restart Thunderbird again... then it is gone again.

I've got a support ticket in at Mozilla now.  I'm starting to think this is a Thunderbird issue.

**Comment by RealRaven2000** [comment]
interesting
 can you reset error console, just before you create the account.  then do
the step you describe and capture the log immediately afterwards?   it
sounds like the code that iterates the accounts is stuck and becomes
"unstuck" when you create the account; if my theory is right we should see
all the missing messages from QF immediately after you made the account...

do you use any addons that create special accounts such as localfolders or
owl?

**Comment by amfonda** [comment]
I do not use any addons that create special accounts such as localfolders or owl.

OK, I recreated the sequence I described above, capturing the debug output, first deleting anything, then running it and copying everything produced:

```
QuickFolders: add toggle-foldertree command... qf-background.js:600:13
notifyTools.registerListener() has been deprecated, use notifyTools.addListener() instead. notifyTools.js:154:13
QuickFolders [logTime init]
 ============INJECT==========
qf-messenger.js onLoad(true) quickfolders-util.js:1304:13
WebExtensions: Adding QuickFolders toolbar ... qf-messenger.js:38
QuickFolders 4:38:29.808  [5 ms]   
 Adding Folder Listener... quickfolders-util.js:1304:13
notifyTools.registerListener() has been deprecated, use notifyTools.addListener() instead. notifyTools.js:154:13
WebExtensions: qf-3pane.js - onLoad() qf-3pane.js:57
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
QuickFolders 4:38:29.887  [79 ms]   
 restore categories from tab session quickfolders-util.js:1304:13
QuickFolders readTabCategorySession() quickfolders-util.js:1315:12
WebExtensions: qf-3pane.js - onLoad() qf-3pane.js:57
QuickFolders 4:38:29.896  [9 ms]   
 readTabCategorySession - for tabId[1] returned __ALL quickfolders-util.js:1304:13
QuickFolders readTabToolbarSession() quickfolders-util.js:1315:12
QuickFolders 4:38:29.897  [1 ms]   
 readTabToolbarSession - for tabId[1] returned {"mainVisibleState":true} quickfolders-util.js:1304:13
QuickFolders 4:38:29.898  [1 ms]   
 initDocAndWindow()
QuickFolders_globalDoc = chrome://messenger/content/messenger.xhtml quickfolders-util.js:1304:13
QuickFolders 4:38:29.898  [0 ms]   
 initDelayed ==== correct window: chrome://messenger/content/messenger.xhtml - Account Settings - Mozilla Thunderbird
wait 10000 msec until init()... quickfolders-util.js:1304:13
QuickFolders 4:38:29.898  [0 ms]   
 Adding Search Input event handler... quickfolders-util.js:1304:13
getFiltersBoolPref(autoStart) failed
defaulting to undefined
Component returned failure code: 0x8000ffff (NS_ERROR_UNEXPECTED) [nsIPrefBranch.getBoolPref]
 aSourceName: ?, line: 427, copl: 0, flg: 1 getFiltersBoolPref
  chrome://quickfolders/content/quickfolders-preferences.js:427:27
initDelayed@chrome://quickfolders/content/quickfolders.js:600:17
quickfolders-util.js:1330:15
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 8 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
WebExtensions: QuickFolders: injecting current folder qf-3pane.js:70
QuickFolders 4:38:30.846  [948 ms]   
 ============INJECT==========
qf-3pane.js onLoad(true) quickfolders-util.js:1304:13
QuickFolders 4:38:30.848  [2 ms]   
 liftNavigationbar() quickfolders-util.js:1304:13
QuickFolders 4:38:30.852  [4 ms]   
 ensureStyleSheetLoaded() - missing style sheet 'quickfolders-layout.css' - not found
 not attempting any style modifications. 
HTMLDocument about:3pane
quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 4:38:30.853  [1 ms]   
 Current Folder Bar - Collapsing optional Navigation Elements:
hideMsgNavigation=false
hideFolderNavigation=false
quickfolders-util.js:1299:13
QuickFolders 4:38:30.898  [45 ms]   
 ensureStyleSheetLoaded() - missing style sheet 'quickfolders-layout.css' - not found
 not attempting any style modifications. 
HTMLDocument about:3pane
quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 4:38:30.898  [0 ms]   
 Current Folder Bar - Collapsing optional Navigation Elements:
hideMsgNavigation=false
hideFolderNavigation=false
quickfolders-util.js:1299:13
QuickFolders 4:38:30.918  [20 ms]   
 read() - Failure: NotFoundError: Could not open the file at C:\Users\mark\AppData\Roaming\Thunderbird\Profiles\js78odf6.default-release\extensions\quickFolders-FolderTree.json quickfolders-util.js:1304:13
QuickFolders.FolderTree.loadDictionary() - Inbox-amfonda@yahoo.com
Component returned failure code: 0x80550005 [nsIMsgFolder.getStringProperty]
 aSourceName: ?, line: 215, copl: 0, flg: 1 loadDictionary
  chrome://quickfolders/content/quickfolders-folderTree.js:215:26
quickfolders-util.js:1330:15
QuickFolders.FolderTree.loadDictionary() - Inbox-mfonda@MegiCorp.com
Component returned failure code: 0x80550005 [nsIMsgFolder.getStringProperty]
 aSourceName: ?, line: 215, copl: 0, flg: 1 loadDictionary
  chrome://quickfolders/content/quickfolders-folderTree.js:215:26
quickfolders-util.js:1330:15
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
QuickFolders 4:38:39.900  [8982 ms]   
 Version() getter. addonInfo: 
Object { id: "quickfolders@curious.be", name: "QuickFolders", description: "Bookmark your favorite mail folders in Thunderbird.", version: "6.3.1pre2", mayDisable: true, enabled: true, optionsUrl: "moz-extension://6efffd4c-9e13-4a84-968f-335177f96924/html/options.html", installType: "normal", type: "extension", permissions: (8) […], … }
quickfolders-util.js:1304:13
QuickFolders 4:38:39.901  [1 ms]   
 Wrapping renameFolder... quickfolders-util.js:1304:13
QuickFolders:QuickFolders.init() - QuickFolders Version 6.3.1pre2
Running on Thunderbird Version 115.4.2
QuickFolders 4:38:39.902  [1 ms]   
 initTabsFromEntries() quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 4:38:39.902  [0 ms]   
 updateUserStyles() chrome://messenger/content/messenger.xhtml quickfolders-util.js:1299:13
QuickFolders 4:38:39.922  [20 ms]   
 init: setting categories to __ALL quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 4:38:39.923  [1 ms]   
 type: contentTab
name: contentTab
tabType: [object Object]
tabs: [object Object]
quickfolders-util.js:1299:13
QuickFolders {INTERFACE} 4:38:39.924  [1 ms]   
 mode = contentTab
isMailPanel = false quickfolders-util.js:1299:13
QuickFolders 4:38:39.924  [0 ms]   
 QuickFolders.init() ends. quickfolders-util.js:1304:13
QuickFolders 4:38:39.924  [0 ms]   
 bookmarks.load… quickfolders-util.js:1304:13
QuickFolders 4:38:39.924  [0 ms]   
 initLicensedUI - hasLicense = true
 licenseInfo: 
Object { status: "Valid", description: "Valid", licensedDaysLeft: 104, expiredDays: 0, expiryDate: "2024-02-21", email: "mark@fonda.org", licenseKey: "QF-mark@fonda.org:2024-02-21;1ac742db92a7fabb5727c514aa9ea13cc6a145bb9e3f6ab5dabf189876c7ec03", decryptedPart: "QF-mark@fonda.org:2024-02-21", keyType: 0, isValid: true, … }
quickfolders-util.js:1304:13
QuickFolders 4:38:39.925  [1 ms]   
 License found - removing Animations()... quickfolders-util.js:1304:13
QuickFolders 4:38:39.927  [2 ms]   
 updateQuickFoldersLabel() quickfolders-util.js:1304:13
QuickFolders 4:38:39.927  [0 ms]   
 updateMainWindow()... quickfolders-util.js:1304:13
QuickFolders 4:38:39.928  [1 ms]   
 bookmarks.load… quickfolders-util.js:1304:13
QuickFolders 4:38:39.928  [0 ms]   
 bookmarks.load complete. quickfolders-util.js:1304:13
QuickFolders 4:38:39.986  [58 ms]   
 updateMainWindow()... quickfolders-util.js:1304:13
QuickFolders 4:38:39.986  [0 ms]   
 bookmarks.load… quickfolders-util.js:1304:13
QuickFolders 4:38:39.986  [0 ms]   
 bookmarks.load complete. quickfolders-util.js:1304:13
QuickFolders 4:38:39.987  [1 ms]   
 readStringFile() - no data - file doesn't exist quickfolders-util.js:1304:13
QuickFolders 4:38:39.987  [0 ms]   
 readStringFile() - no data - file doesn't exist quickfolders-util.js:1304:13
QuickFolders 4:38:39.988  [1 ms]   
 updateFoldersUI()... quickfolders-util.js:1304:13
Starting updateFolders - Active Categories: quickfolders-interface.js:660:15
QuickFolders {INTERFACE} 4:38:39.988  [0 ms]   
 updateFolders(rebuildCategories=true, minimalUpdate=false) quickfolders-util.js:1299:13
QuickFolders 4:38:39.988  [0 ms]   
 updateFolders(rebuildCategories: true, minimal: false) - Number of Folders = 97 quickfolders-util.js:1304:13
QuickFolders 4:38:39.988  [0 ms]   
 updateQuickFoldersLabel() quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 4:38:39.989  [1 ms]   
 updateCategories() quickfolders-util.js:1299:13
QuickFolders 4:38:39.989  [0 ms]   
 updateCategories() - [0 Categories] quickfolders-util.js:1304:13
QuickFolders 4:38:39.989  [0 ms]   
 No Categories defined, hiding Categories box. quickfolders-util.js:1304:13
QuickFolders 4:38:39.989  [0 ms]   
 Set category for tab [1] to __ALL quickfolders-util.js:1304:13
QuickFolders 4:38:40.3  [14 ms]   
 97 of 97 tabs rebuilt [minimalUpdate=false]. quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 4:38:40.3  [0 ms]   
 onTabSelected()
 folder = <none> quickfolders-util.js:1299:13
QuickFolders.Interface.updateFolders(rebuildCat = true, minimal = false)
 97 folders took: [15 ms] quickfolders-interface.js:817:15
QuickFolders {INTERFACE} 4:38:40.4  [1 ms]   
 updateUserStyles() chrome://messenger/content/messenger.xhtml quickfolders-util.js:1299:13
QuickFolders 4:38:40.43  [39 ms]   
 readStringFile() - no data - file doesn't exist quickfolders-util.js:1304:13
QuickFolders 4:38:40.44  [1 ms]   
 updateFoldersUI()... quickfolders-util.js:1304:13
Starting updateFolders - Active Categories: __ALL quickfolders-interface.js:660:15
QuickFolders {INTERFACE} 4:38:40.44  [0 ms]   
 updateFolders(rebuildCategories=true, minimalUpdate=false) quickfolders-util.js:1299:13
QuickFolders 4:38:40.44  [0 ms]   
 updateFolders(rebuildCategories: true, minimal: false) - Number of Folders = 97 quickfolders-util.js:1304:13
QuickFolders 4:38:40.45  [1 ms]   
 updateQuickFoldersLabel() quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 4:38:40.46  [1 ms]   
 updateCategories() quickfolders-util.js:1299:13
QuickFolders 4:38:40.46  [0 ms]   
 updateCategories() - [0 Categories] quickfolders-util.js:1304:13
QuickFolders 4:38:40.46  [0 ms]   
 No Categories defined, hiding Categories box. quickfolders-util.js:1304:13
QuickFolders 4:38:40.46  [0 ms]   
 Set category for tab [1] to __ALL quickfolders-util.js:1304:13
QuickFolders 4:38:40.56  [10 ms]   
 97 of 97 tabs rebuilt [minimalUpdate=false]. quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 4:38:40.56  [0 ms]   
 onTabSelected()
 folder = <none> quickfolders-util.js:1299:13
QuickFolders.Interface.updateFolders(rebuildCat = true, minimal = false)
 97 folders took: [12 ms] quickfolders-interface.js:817:15
QuickFolders {INTERFACE} 4:38:40.57  [1 ms]   
 updateUserStyles() chrome://messenger/content/messenger.xhtml quickfolders-util.js:1299:13
This site appears to use a scroll-linked positioning effect. This may not work well with asynchronous panning; see https://firefox-source-docs.mozilla.org/performance/scroll-linked_effects.html for further details and to join the discussion on related tools and features! index.html

```

**Comment by RealRaven2000** [comment]
thanks. not a single message here is from QuickFolders, if you disable thje two points "info and debug" but leave errors, warnings and logs enabled, does it look the same?

**Comment by amfonda** [comment]
I tried again, with the settings you specified.  Here's the debug output:

```
QuickFolders: add toggle-foldertree command... qf-background.js:600:13
notifyTools.registerListener() has been deprecated, use notifyTools.addListener() instead. notifyTools.js:154:13
QuickFolders [logTime init]
 ============INJECT==========
qf-messenger.js onLoad(true) quickfolders-util.js:1304:13
WebExtensions: Adding QuickFolders toolbar ... qf-messenger.js:38
QuickFolders 15:58:6.704  [5 ms]   
 Adding Folder Listener... quickfolders-util.js:1304:13
notifyTools.registerListener() has been deprecated, use notifyTools.addListener() instead. notifyTools.js:154:13
WebExtensions: qf-3pane.js - onLoad() 2 qf-3pane.js:57
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
QuickFolders 15:58:6.870  [166 ms]   
 restore categories from tab session quickfolders-util.js:1304:13
QuickFolders readTabCategorySession() quickfolders-util.js:1315:12
QuickFolders 15:58:6.892  [22 ms]   
 readTabCategorySession - for tabId[2] returned __ALL quickfolders-util.js:1304:13
QuickFolders readTabToolbarSession() quickfolders-util.js:1315:12
QuickFolders 15:58:6.905  [13 ms]   
 readTabToolbarSession - for tabId[2] returned {"mainVisibleState":true} quickfolders-util.js:1304:13
QuickFolders 15:58:6.905  [0 ms]   
 initDocAndWindow()
QuickFolders_globalDoc = chrome://messenger/content/messenger.xhtml quickfolders-util.js:1304:13
QuickFolders 15:58:6.905  [0 ms]   
 initDelayed ==== correct window: chrome://messenger/content/messenger.xhtml - Calendar - Mozilla Thunderbird
wait 10000 msec until init()... quickfolders-util.js:1304:13
QuickFolders 15:58:6.905  [0 ms]   
 Adding Search Input event handler... quickfolders-util.js:1304:13
getFiltersBoolPref(autoStart) failed
defaulting to undefined
Component returned failure code: 0x8000ffff (NS_ERROR_UNEXPECTED) [nsIPrefBranch.getBoolPref]
 aSourceName: ?, line: 427, copl: 0, flg: 1 getFiltersBoolPref
  chrome://quickfolders/content/quickfolders-preferences.js:427:27
initDelayed@chrome://quickfolders/content/quickfolders.js:600:17
quickfolders-util.js:1330:15
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 8 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
WebExtensions: QuickFolders: injecting current folder qf-3pane.js:70
QuickFolders 15:58:7.749  [844 ms]   
 ============INJECT==========
qf-3pane.js onLoad(true) quickfolders-util.js:1304:13
QuickFolders 15:58:7.751  [2 ms]   
 liftNavigationbar() quickfolders-util.js:1304:13
QuickFolders 15:58:7.755  [4 ms]   
 ensureStyleSheetLoaded() - missing style sheet 'quickfolders-layout.css' - not found
 not attempting any style modifications. 
HTMLDocument about:3pane
quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 15:58:7.755  [0 ms]   
 Current Folder Bar - Collapsing optional Navigation Elements:
hideMsgNavigation=false
hideFolderNavigation=false
quickfolders-util.js:1299:13
QuickFolders 15:58:7.800  [45 ms]   
 ensureStyleSheetLoaded() - missing style sheet 'quickfolders-layout.css' - not found
 not attempting any style modifications. 
HTMLDocument about:3pane
quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 15:58:7.800  [0 ms]   
 Current Folder Bar - Collapsing optional Navigation Elements:
hideMsgNavigation=false
hideFolderNavigation=false
quickfolders-util.js:1299:13
QuickFolders 15:58:7.837  [37 ms]   
 read() - Failure: NotFoundError: Could not open the file at C:\Users\mark\AppData\Roaming\Thunderbird\Profiles\js78odf6.default-release\extensions\quickFolders-FolderTree.json quickfolders-util.js:1304:13
QuickFolders.FolderTree.loadDictionary() - Inbox-amfonda@yahoo.com
Component returned failure code: 0x80550005 [nsIMsgFolder.getStringProperty]
 aSourceName: ?, line: 215, copl: 0, flg: 1 loadDictionary
  chrome://quickfolders/content/quickfolders-folderTree.js:215:26
quickfolders-util.js:1330:15
QuickFolders.FolderTree.loadDictionary() - Inbox-mfonda@MegiCorp.com
Component returned failure code: 0x80550005 [nsIMsgFolder.getStringProperty]
 aSourceName: ?, line: 215, copl: 0, flg: 1 loadDictionary
  chrome://quickfolders/content/quickfolders-folderTree.js:215:26
quickfolders-util.js:1330:15
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 2 crypto-SDR.sys.mjs:199
QuickFolders 15:58:16.908  [9071 ms]   
 Version() getter. addonInfo: 
Object { id: "quickfolders@curious.be", name: "QuickFolders", description: "Bookmark your favorite mail folders in Thunderbird.", version: "6.3.1pre2", mayDisable: true, enabled: true, optionsUrl: "moz-extension://6efffd4c-9e13-4a84-968f-335177f96924/html/options.html", installType: "normal", type: "extension", permissions: (8) […], … }
quickfolders-util.js:1304:13
QuickFolders 15:58:16.911  [3 ms]   
 Wrapping renameFolder... quickfolders-util.js:1304:13
QuickFolders:QuickFolders.init() - QuickFolders Version 6.3.1pre2
Running on Thunderbird Version 115.4.2
QuickFolders 15:58:16.912  [1 ms]   
 initTabsFromEntries() quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 15:58:16.913  [1 ms]   
 updateUserStyles() chrome://messenger/content/messenger.xhtml quickfolders-util.js:1299:13
QuickFolders 15:58:16.933  [20 ms]   
 init: setting categories to __ALL quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 15:58:16.934  [1 ms]   
 type: calendar
maxTabs: 1
openTab: openTab(tab)
showTab: showTab(tab)
closeTab: closeTab(tab)
persistTab: persistTab(tab)
restoreTab: restoreTab(tabmail, state)
onTitleChanged: onTitleChanged(tab)
supportsCommand: (aCommand, aTab)
isCommandEnabled: (aCommand, aTab)
doCommand: (aCommand, aTab)
onEvent: (aEvent, aTab)
notificationbox: [object Object]
name: calendar
tabType: [object Object]
tabs: [object Object]
quickfolders-util.js:1299:13
QuickFolders {INTERFACE} 15:58:16.934  [0 ms]   
 mode = calendar
isMailPanel = false quickfolders-util.js:1299:13
QuickFolders 15:58:16.935  [1 ms]   
 QuickFolders.init() ends. quickfolders-util.js:1304:13
QuickFolders 15:58:16.935  [0 ms]   
 bookmarks.load… quickfolders-util.js:1304:13
QuickFolders 15:58:16.935  [0 ms]   
 initLicensedUI - hasLicense = true
 licenseInfo: 
Object { status: "Valid", description: "Valid", licensedDaysLeft: 102, expiredDays: 0, expiryDate: "2024-02-21", email: "mark@fonda.org", licenseKey: "QF-mark@fonda.org:2024-02-21;1ac742db92a7fabb5727c514aa9ea13cc6a145bb9e3f6ab5dabf189876c7ec03", decryptedPart: "QF-mark@fonda.org:2024-02-21", keyType: 0, isValid: true, … }
quickfolders-util.js:1304:13
QuickFolders 15:58:16.935  [0 ms]   
 License found - removing Animations()... quickfolders-util.js:1304:13
QuickFolders 15:58:16.937  [2 ms]   
 updateQuickFoldersLabel() quickfolders-util.js:1304:13
QuickFolders 15:58:16.938  [1 ms]   
 updateMainWindow()... quickfolders-util.js:1304:13
QuickFolders 15:58:16.938  [0 ms]   
 bookmarks.load… quickfolders-util.js:1304:13
QuickFolders 15:58:16.938  [0 ms]   
 bookmarks.load complete. quickfolders-util.js:1304:13
QuickFolders 15:58:16.998  [60 ms]   
 updateMainWindow()... quickfolders-util.js:1304:13
QuickFolders 15:58:16.999  [1 ms]   
 bookmarks.load… quickfolders-util.js:1304:13
QuickFolders 15:58:16.999  [0 ms]   
 bookmarks.load complete. quickfolders-util.js:1304:13
QuickFolders 15:58:17.0  [1 ms]   
 readStringFile() - no data - file doesn't exist quickfolders-util.js:1304:13
QuickFolders 15:58:17.1  [1 ms]   
 readStringFile() - no data - file doesn't exist quickfolders-util.js:1304:13
QuickFolders 15:58:17.2  [1 ms]   
 updateFoldersUI()... quickfolders-util.js:1304:13
Starting updateFolders - Active Categories: quickfolders-interface.js:660:15
QuickFolders {INTERFACE} 15:58:17.2  [0 ms]   
 updateFolders(rebuildCategories=true, minimalUpdate=false) quickfolders-util.js:1299:13
QuickFolders 15:58:17.2  [0 ms]   
 updateFolders(rebuildCategories: true, minimal: false) - Number of Folders = 87 quickfolders-util.js:1304:13
QuickFolders 15:58:17.2  [0 ms]   
 updateQuickFoldersLabel() quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 15:58:17.2  [0 ms]   
 updateCategories() quickfolders-util.js:1299:13
QuickFolders 15:58:17.3  [1 ms]   
 updateCategories() - [0 Categories] quickfolders-util.js:1304:13
QuickFolders 15:58:17.3  [0 ms]   
 No Categories defined, hiding Categories box. quickfolders-util.js:1304:13
QuickFolders 15:58:17.3  [0 ms]   
 Set category for tab [4] to __ALL quickfolders-util.js:1304:13
QuickFolders 15:58:17.15  [12 ms]   
 87 of 87 tabs rebuilt [minimalUpdate=false]. quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 15:58:17.15  [0 ms]   
 onTabSelected()
 folder = <none> quickfolders-util.js:1299:13
QuickFolders.Interface.updateFolders(rebuildCat = true, minimal = false)
 87 folders took: [13 ms] quickfolders-interface.js:817:15
QuickFolders {INTERFACE} 15:58:17.15  [0 ms]   
 updateUserStyles() chrome://messenger/content/messenger.xhtml quickfolders-util.js:1299:13
QuickFolders 15:58:17.68  [53 ms]   
 readStringFile() - no data - file doesn't exist quickfolders-util.js:1304:13
QuickFolders 15:58:17.69  [1 ms]   
 updateFoldersUI()... quickfolders-util.js:1304:13
Starting updateFolders - Active Categories: __ALL quickfolders-interface.js:660:15
QuickFolders {INTERFACE} 15:58:17.69  [0 ms]   
 updateFolders(rebuildCategories=true, minimalUpdate=false) quickfolders-util.js:1299:13
QuickFolders 15:58:17.69  [0 ms]   
 updateFolders(rebuildCategories: true, minimal: false) - Number of Folders = 87 quickfolders-util.js:1304:13
QuickFolders 15:58:17.70  [1 ms]   
 updateQuickFoldersLabel() quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 15:58:17.71  [1 ms]   
 updateCategories() quickfolders-util.js:1299:13
QuickFolders 15:58:17.71  [0 ms]   
 updateCategories() - [0 Categories] quickfolders-util.js:1304:13
QuickFolders 15:58:17.71  [0 ms]   
 No Categories defined, hiding Categories box. quickfolders-util.js:1304:13
QuickFolders 15:58:17.71  [0 ms]   
 Set category for tab [4] to __ALL quickfolders-util.js:1304:13
QuickFolders 15:58:17.80  [9 ms]   
 87 of 87 tabs rebuilt [minimalUpdate=false]. quickfolders-util.js:1304:13
QuickFolders {INTERFACE} 15:58:17.80  [0 ms]   
 onTabSelected()
 folder = <none> quickfolders-util.js:1299:13
QuickFolders.Interface.updateFolders(rebuildCat = true, minimal = false)
 87 folders took: [11 ms] quickfolders-interface.js:817:15
QuickFolders {INTERFACE} 15:58:17.80  [0 ms]   
 updateUserStyles() chrome://messenger/content/messenger.xhtml quickfolders-util.js:1299:13
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
NS_ERROR_FAILURE: Couldn't decrypt string 3 crypto-SDR.sys.mjs:199
This site appears to use a scroll-linked positioning effect. This may not work well with asynchronous panning; see https://firefox-source-docs.mozilla.org/performance/scroll-linked_effects.html for further details and to join the discussion on related tools and features! index.html

```

**Comment by amfonda** [comment]
Could this issue have something to do with the prefs.js file in my Thunderbird profile directory?

I have one other issue with my upgrade to TB 115 which is that my local calendars disappeared and I cannot add any new local calendars that reside on my computer.  Remote calendars via internet connections work fine.  So I was troubleshooting this and I found a forum message that resolved a similar problem by restoring an older prefs.js file from a previous profile.  

So I tried using the prefs.js file from my TB 91.13 profile (copying over while TB is closed) and unfortunately it did not solve my calendar issue AND it also disabled my workaround for QuickFolders.  I've been able to invoke QF by 'faking' the addition of a new mail account (I cancel before adding anything), but the old prefs.js file would not allow me to do that.  

So I reloaded the newer prefs.js file (created under TB 115) and it worked again.  I've examined the prefs.js file and it has all of my TB settings including add-ons such as QF.  Could there be something in there that needs fixing?  I noticed that most/all of the  contents of prefs.js is also accessible through Config Editor which is probably a safer way to modify prefs.js.  What do you think?

**Edit: Now I realize that the old prefs.js would never work because it has all the addresses and codes for my old setup, not the new one.  Too much has changed.  Good idea, but not much prospect of working.**

**Comment by amfonda** [comment]
Also, I did not upgrade directly from 91.13 to 115.4... I first upgraded to TB 102... which was a disaster until I then tried 115.2 which resolved most of the issues.  I have seen that many people had problems with 102, and in fact I had delayed upgrading because a couple of attempts rendered my whole TB workspace unrecognizable and I reverted back.  Finally I forged ahead once I saw there was a new release in TB 115.  Could there be some fragments of code or preferences that screwed up my profile?

**Edit: No, I don't think this was the issue.  I created a new profile and loaded it with my old TB 91.13 folder contents.  Then I switched to that profile, with TB 115.4.2 installed, and it came up into the same state as my newer profile... i.e. - the same two issues are present (Missing Calendars and QF Not Starting). So my conclusion is that its a TB 115 issue, nothing to do with the TB 102 intermediate step.**

**Comment by RealRaven2000** [comment]
Still a bit stumped by this one, even when reading your log - the toolbar not displaying at all seems to point to an issue with the WindowListener / WindowLoader - which injects the xhtml into the main document. Either injection fails or something makes the toolbar invisible. Just remind me, do you see the item in the view menu?

![image](https://github.com/RealRaven2000/QuickFolders/assets/6314674/0cb8be4e-3961-421d-b4bb-08d793f134f0)

As far as I remember, you don't see this at all so it's not invisible but really not present at all?

**Comment by amfonda** [comment]
Yes, correct... upon starting TB, the QuickFolder Toolbar item is not visible in the View menu.  It becomes visible only after I do the Add Mail Account trick.  Immediately after doing that, the QF toolbar appears in the View menu and the actual QF toolbar starts initializing and then appears on its own after about 5 seconds.  I don't need to select it in the View menu, it appears on its own, while I am viewing the mail tab.

**Comment by amfonda** [comment]
Still no resolution to this bug.  I'm using the workaround each time I open Thunderbird where I start to add a new mail account then cancel, which invokes Quickfolders to initialize.  Hoping that somehow an update to either TB or QF will clear this bug.

**Comment by amfonda** [comment]
Still no resolution on this issue.  I continue to use a workaround to initilize Quickfolders each time I open Thunderbird (Account Settings > Account Actions > Add Mail Account > Cancel).  I'm using TB 115.11.1 and QF 6.6.1 as per normal updates.

I see the latest QF 6.6.1 update has a note: "Known Issue: The verification of QuickFolders licenses currently doesn't work when the license is bound to an exchange account when it is accessed using Owl. This is likely due to a bug in the web extensions API."  I believe this is the same faulty license access issue and await eventual resolution.

I have no idea what Owl is and if I am bound to an exchange account.  I do not believe this applies to my setup.  How do I troubleshoot this?

**Comment by RealRaven2000** [comment]
I suggest we do a Zoom call when I am back from my annual leave - so that would be possible from August 1st. Please suggest a day and time and reach out to me via email to arrange detail.

You can use the support tab on QF settings to write to me directly:

![image](https://github.com/user-attachments/assets/600354ca-477e-4bbd-932e-01574e45cce8)

**Comment by amfonda** [comment]
As you suggested, I created a new TB profile and imported all my data and installed all my add-ons, including QF 6.10.4...

and QuickFolders loaded right up and is working perfectly... other than some minor formatting which I am fixing.

So it seems that the culprit was a corrupted TB profile, as you suspected.  I don't think the QF 6.10.5.pre1 patch is needed.

Thanks so much for your time and persistence!

**Comment by RealRaven2000** [comment]
> As you suggested, I created a new TB profile and imported all my data and installed all my add-ons, including QF 6.10.4...
> and QuickFolders loaded right up and is working perfectly... other than some minor formatting which I am fixing.
> So it seems that the culprit was a corrupted TB profile, as you suspected. I don't think the QF 6.10.5.pre1 patch is needed.

nice one, yeah unfortunately these profiles break sometimes, glad it got resolved. I will close the issue as resolved.
